OpenSim/Actuators/Millard2012EquilibriumMuscle.cpp

```diff
@@ -156,6 +156,12 @@ void Millard2012EquilibriumMuscle::extendFinalizeFromProperties()
     penMdl.set_optimal_fiber_length(getOptimalFiberLength());
     penMdl.set_pennation_angle_at_optimal(getPennationAngleAtOptimalFiberLength());
     penMdl.set_maximum_pennation_angle(get_maximum_pennation_angle());
+    try {
+        penMdl.finalizeFromProperties();
+    } catch (const SimTK::Exception::Base& ex) {
+        penMdl.setDefaultProperties(); //avoid throwing again
```

Comment:
Ideally, we would restore the pendMdl to its previous (working) state and not all the way back to the default values. Assuming that the pendMdl is initialized with default values the effect would be the same until the user define other valid values and then entered an invalid value for one property, which would then reset all the property values.

I liked your idea yesterday of keeping a clone of the subcomponent (pendMdl) and the restoring using assignment which will copy property values, i.e. `pendMdl = *clonePendMdl;`

Reply:
> Ideally, we would restore the pendMdl to its previous (working) state and not all the way back to the default values. Assuming that the pendMdl is initialized with default values the effect would be the same until the user define other valid values and then entered an invalid value for one property, which would then reset all the property values.

`Model::finalizeFromProperties()` won't complete successfully until the Muscle's `optimal_fiber_length` property has been set to a valid quantity. The underlying pennation model subcomponent is a slave of the Muscle and won't be used for anything until a valid optimal fiber length has been set. Thus, the state of the subcomponent is irrelevant and here I'm simply setting its properties to something "safe" to prevent it from throwing. I'll redo it the other way.

Reply:
Good point. Thanks for clarifying. You can just add a comment to that effect without redoing.